dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/logging/LogUtils.java

```diff
@@ -296,14 +320,28 @@ protected static Logger createLogger(final Class<?> cls,
                     throw new RuntimeException(e);
                 }
             }
-
             Logger logger;
             try {
                 logger = Logger.getLogger(loggerName, bundleName); //NOPMD
             } catch (IllegalArgumentException | MissingResourceException ex) {
                 //likely a mismatch on the bundle name, just return the default
                 logger = Logger.getLogger(loggerName); //NOPMD
             }
+            // Liberty change begin: Swap WsLogger with CXFLogger if no logger class name provided
+            if (logger.getClass().equals(wsLoggerClass)) {
+                try {
+                    backupLoggerClass(CXFLogger.class);
+                    return CXFLogger.getLogger(logger, cls);
+                } catch (Exception e) {
+                    // Let system return the logger that is already created
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                        Tr.debug(tc, "Exception occured at CXFLogger.getlogger method: " + e.getMessage() + "\n We'll fallback to original logger class.");
+                    }
+                }
+            } 
+            // If we can't create CXFLogger for any reason restore back the original logger
+            restoreLoggerClass();
```

Comment:
Should this restoreLoggerClass() be moved up inside the if statement? restoreLogger could be called without backupLogger being called first.